src/hotspot/share/code/codeCache.cpp

```diff
@@ -234,6 +236,9 @@ void CodeCache::initialize_heaps() {
   if (!non_nmethod_set && !profiled_set && !non_profiled_set) {
     // Check if we have enough space for the non-nmethod code heap
     if (cache_size > non_nmethod_size) {
+      // Ensure at least min_size is available for profiled and non-profiled
+      // methods
+      non_nmethod_size = MIN2(non_nmethod_size, cache_size - 2 * min_size);
```

Comment:
Would it not be simpler to just do above:

```
if (cache_size > (non_method_size - 2 * min_size)) {
```

?

Also, can this calculation produce 0 or negative overflow?

Reply:
I have changed it a bit now (based on your suggested simplification); please check and see if you agree with the new version. I have also experimented a bit with different flags (`-XX:ReservedCodeCacheSize`, `-XX:InitialCodeCacheSize`, `-XX:+SegmentedCodeCache`, `-XX:CodeCacheMinimumUseSpace`) trying to get the expression `cache_size - 2 * min_size` to negatively overflow, without success. The expression is also present even before my change, so i believe it is safe to assume that `cache_size > 2 * min_size` (but I cannot guarantee it).